src/Pim/Component/Catalog/Normalizer/Storage/Product/ProductValuesNormalizer.php

```diff
@@ -23,7 +23,13 @@ public function normalize($values, $format = null, array $context = [])
         $result = [];
         foreach ($values as $value) {
             $normalizedValue = $this->serializer->normalize($value, $format, $context);
-            $result = array_merge_recursive($result, $normalizedValue);
+            $attributeCode = $value->getAttribute()->getCode();
```

Comment:
Nop, because in the indexing normalizer, your array key will be '123-varchar', not just '123', so no risk it is transformed in numeric.